com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceInstanceGetter.java

```diff
@@ -53,13 +59,11 @@ public ServiceInstanceGetter(RestTemplateFactory restTemplateFactory) {
         return JsonUtil.convertJsonToMap(response);
     }
 
-    @SuppressWarnings("unchecked")
     private Map<String, Object> getCloudServiceInstance(Map<String, Object> serviceInstancesResponse) {
         validateServiceInstanceResponse(serviceInstancesResponse);
         List<Map<String, Object>> cloudServiceInstanceResources = getResourcesFromResponse(serviceInstancesResponse);
         if (!cloudServiceInstanceResources.isEmpty()) {
-            return (Map<String, Object>) cloudServiceInstanceResources.get(0)
-                .get("entity");
+            return cloudServiceInstanceResources.get(0);
```

Comment:
By this, you are returning the whole response from the Controller. Why not getting just the entity?
Just to remote the SuppressWarning?